### PR TITLE
Dtd 217

### PR DESCRIPTION
**Pull request by kiranpl**

### CODE REVIEW

File: src/test/resources/features/ifs/InterestRateChanges.feature [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      | 0                    | 0                    | 500000            | 500000              | 500000              |
+    And the 1st debt summary will contain
+      | interestDueDailyAccrual | interestDueDebtTotal | unpaidAmountDebt | totalAmountIntDebt | numberChargeableDays | amountOnIntDueDebt |
+      | 0                       | 0                    | 500000           | 500000             | 0                    | 500000             |
+    And the 1st debt summary will have calculation windows
+      | periodFrom | periodTo   | numberOfDays | interestRate | interestDueDailyAccrual | amountOnIntDueWindow | interestDueWindow | unpaidAmountWindow |
+      | 2000-01-01 | 2000-02-05 | 0            | 0.0          | 0                       | 500000               | 0                 | 500000             |
+      | 2000-02-06 | 2001-03-31 | 418          | 8.5          | 13                      | 500904               | 780               | 500780             |
+
+  #Scenario: Interest rate changes from 1% to 0%
```
Comment (rogerball) [inline src/test/resources/features/ifs/InterestRateChanges.feature]: Does this scenario need to be added?

Comment (kiranpl) [reply src/test/resources/features/ifs/InterestRateChanges.feature]: Yes, currently there is no way to implement it based on the test data available

**Review by thadeusssewanyana — APPROVED** [review]

File: src/test/resources/features/ifs/InterestRateChanges.feature [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    And the 2nd debt summary will have calculation windows
+      | periodFrom | periodTo   | numberOfDays | interestRate | interestDueDailyAccrual | interestDueWindow | unpaidAmountWindow | amountOnIntDueWindow |
+      | 2020-01-16 | 2020-03-29 | 74           | 3.25         | 44                      | 3294              | 500000             | 500000               |
+      | 2020-03-30 | 2020-04-06 | 8            | 2.75         | 37                      | 301               | 500000             | 500000               |
+      | 2020-04-07 | 2021-01-19 | 288          | 2.6          | 35                      | 10257             | 500000             | 500000               |
+      | 2021-01-20 | 2021-03-09 | 49           | 2.6          | 28                      | 1396              | 400000             | 400000               |
+      | 2021-03-10 | 2021-04-14 | 36           | 2.6          | 21                      | 769               | 300000             | 300000               |
+
+  Scenario: Interest rate changes from 2.75% to 2.6% - dateCalculationTo before interestStartDate
```
Comment (rogerball) [inline src/test/resources/features/ifs/InterestRateChanges.feature:117]: Interest rate used in the scenario is 3.25 and dateCalculationTo is after interestStartDate

File: src/test/resources/features/ifs/InterestRateChanges.feature [hunk 3]
```diff
@@ -0,0 +1,131 @@
+# originalAmount (Amount) = 500,000
+# dateCreated (Date Amount) = 01/01/2020
+# mainTrans (Regime = DRIER) = 1085
```
Comment (rogerball) [inline src/test/resources/features/ifs/InterestRateChanges.feature]: 1525

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# originalAmount (Amount) = 500,000
+# dateCreated (Date Amount) = 01/01/2020
+# mainTrans (Regime = DRIER) = 1085
```
Comment (rogerball) [inline src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature]: 1525

File: src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# originalAmount (Amount) = 500,000
+# dateCreated (Date Amount) = 01/01/2020
+# mainTrans (Regime = DRIER) = 1085
+# subTrans (CHARGE_TYPE == Duty Repaid in Error - National Insurance) = 1025
```
Comment (rogerball) [inline src/test/resources/features/ifs/InterestRateChanges EdgeCases.feature]: 1000 or 1090 used in scenarios

File: src/test/resources/features/ifs/InterestRateChanges.feature [hunk 6]
```diff
@@ -0,0 +1,131 @@
+# originalAmount (Amount) = 500,000
+# dateCreated (Date Amount) = 01/01/2020
+# mainTrans (Regime = DRIER) = 1085
+# subTrans (CHARGE_TYPE == Duty Repaid in Error - National Insurance) = 1025
```
Comment (rogerball) [inline src/test/resources/features/ifs/InterestRateChanges.feature]: 1000 or 1090 used in scenarios

**Review by rogerball — APPROVED** [review]

**Pull request merged**